multitool/private/cwd.bzl

```diff
@@ -1,12 +1,22 @@
 "cwd: a rule for executing an executable in the BUILD_WORKING_DIRECTORY"
 
 def _cwd_impl(ctx):
-    output = ctx.actions.declare_file(ctx.label.name)
+    template = ctx.file._template_sh
+    wrapper_name = ctx.label.name
+    tool_short_path = ctx.file.tool.short_path
+    if ctx.file.tool.extension == "exe":
+        template = ctx.file._template_bat
+        wrapper_name = wrapper_name + ".bat"
+        # with runfiles enabled, same as linux:
+        tool_short_path = tool_short_path.replace("/", "\\")
```

Comment:
I'm really surprised that Bazel doesn't take care of this for you

Reply:
Agreed... does multitool work on linux if runfiles are disabled?

Reply:
Both MacOS and Linux are unaffected by setting `--noenable_runfiles`

Reply:
...runfiles do end up empty
```
tree bazel-bin/integration_test.runfiles/
bazel-bin/integration_test.runfiles/
├── MANIFEST -> <snip>/execroot/_main/bazel-out/darwin_arm64-fastbuild/bin/integration_test.runfiles_manifest
└── _main
```

but MANIFEST contains mappings to the external tools

Reply:
Did you try this without the forward slash to backward slash converstion? [One reference](https://retrocomputing.stackexchange.com/questions/28344/since-when-does-windows-support-forward-slash-as-path-separator#:~:text=The%20DOS%20kernel%2C%20on%20which,filename%20is%20a%20forward%20slash.) seems to suggest forward slashes should be interpreted reasonably.

Reply:
I'm seeing something different. Neither linux nor windows are working when I specify --noenable_runfiles:

This is linux, rules_multitool main, head, bazel 7.0.2 and 7.2.0 both the same:
```
cd examples/module
% bazel run @multitool//tools/target-determinator:cwd --noenable_runfiles
INFO: Analyzed target @@rules_multitool~override~multitool~multitool//tools/target-determinator:cwd (68 packages loaded, 312 targets configured).
INFO: Found 1 target...
Target @@rules_multitool~override~multitool~multitool//tools/target-determinator:cwd up-to-date:
  bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/cwd
INFO: Elapsed time: 13.839s, Critical Path: 0.02s
INFO: 6 processes: 6 internal.
INFO: Build completed successfully, 6 total actions
INFO: Running command line: bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/cwd
/u/chris/.cache/bazel/_bazel_chris/568a4c36a179246e6ffaf6235b50ee36/execroot/_main/bazel-out/k8-fastbuild/bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/cwd: line 4: /u/chris/.cache/bazel/_bazel_chris/568a4c36a179246e6ffaf6235b50ee36/execroot/_main/bazel-out/k8-fastbuild/bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/cwd.runfiles/../rules_multitool~override~multitool~multitool/tools/target-determinator/target-determinator: No such file or directory
```

Reply:
> Did you try this without the forward slash to backward slash converstion?

In this case, it works. The thing is, the program is executed with a mixed path containing backslashes for the first part, and forward slashes for the last part. Programs will see the mixed path as $ARGV[0]; if they inspect their own path may or may not handle the forward slashes; there is no requirement on windows apps to do so. I'd prefer to keep the normalization as a belt-and-braces reliability measure.

Reply:
On main for both of these:

macos:
```
❯ uname -srvmo
Darwin 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64
❯ cd examples/module
❯ bazel clean --expunge
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
❯ bazel run --noenable_runfiles @multitool//tools/target-determinator -- -version
Starting local Bazel server and connecting to it...
INFO: Analyzed target @@rules_multitool~override~multitool~multitool//tools/target-determinator:target-determinator (68 packages loaded, 357 targets configured).
INFO: Found 1 target...
Target @@rules_multitool~override~multitool~multitool//tools/target-determinator:target-determinator up-to-date:
  bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/target-determinator
INFO: Elapsed time: 10.083s, Critical Path: 0.02s
INFO: 5 processes: 5 internal.
INFO: Build completed successfully, 5 total actions
INFO: Running command line: bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/target-determinator -version
target-determinator v0.25.0
```

linux:
```
$ uname -srvmpio
Linux 5.10.216-204.855.amzn2.x86_64 #1 SMP Sat May 4 16:53:27 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
$ cd examples/module
$ bazel clean --expunge
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
$ bazel run @multitool//tools/target-determinator -- -version
Starting local Bazel server and connecting to it...
INFO: Analyzed target @@rules_multitool~override~multitool~multitool//tools/target-determinator:target-determinator (67 packages loaded, 314 targets configured).
INFO: Found 1 target...
Target @@rules_multitool~override~multitool~multitool//tools/target-determinator:target-determinator up-to-date:
  bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/target-determinator
INFO: Elapsed time: 13.833s, Critical Path: 0.04s
INFO: 5 processes: 5 internal.
INFO: Build completed successfully, 5 total actions
INFO: Running command line: bazel-bin/external/rules_multitool~override~multitool~multitool/tools/target-determinator/target-determinator -version
target-determinator v0.25.0
```

Reply:
Ah sorry, you're looking specifically at the cwd targets, those do fail, yes.

Reply:
It's not clear to me how one would make the cwd target work without runfiles, though -- we need the the toolchain to get included somewhere, and DefaultInfo doesn't give us another way to do that aside from runfiles.

Reply:
The .sh would need to parse the manifest file and pick the path from there, I think

Reply:
Ok, then for now I think we'll continue to require runfiles support, that doesn't preclude having an eventual setup that doesn't require it, but that can be a separate change.